### ISSUE TITLE
loss don't decline

### ISSUE THREAD

**Issue by xiaoshuyun** [zylo117/Yet-Another-EfficientDet-Pytorch#242]
I train on my dataset. Firstly,I set lr=e-5 ,batch size=15,after 100 epoch ,the train loss and val loss are no longer falling.val loss  = 0.5 .then I change the lr=e-5.Loss still didnt decline.
Step: 110999. Epoch: 299/500. Iteration: 370/370. Cls loss: 0.54892. Reg loss: 0.07947. Total loss: 0.62839: 100%|█| 370/370 [02:39
Val. Epoch: 299/500. Classification loss: 0.56471. Regression loss: 0.09870. Total loss: 0.66342
Step: 111369. Epoch: 300/500. Iteration: 370/370. Cls loss: 0.50704. Reg loss: 0.07944. Total loss: 0.58648: 100%|█| 370/370 [02:38
Val. Epoch: 300/500. Classification loss: 0.56558. Regression loss: 0.10068. Total loss: 0.66626
Step: 111499. Epoch: 301/500. Iteration: 130/370. Cls loss: 0.52113. Reg loss: 0.08830. Total loss: 0.60943:  35%|▎| 129/370 [00:59checkpoint...
Step: 111739. Epoch: 301/500. Iteration: 370/370. Cls loss: 0.53809. Reg loss: 0.09533. Total loss: 0.63342: 100%|█| 370/370 [02:39
Val. Epoch: 301/500. Classification loss: 0.56361. Regression loss: 0.09652. Total loss: 0.66012
Step: 111999. Epoch: 302/500. Iteration: 260/370. Cls loss: 0.47364. Reg loss: 0.08025. Total loss: 0.55390:  70%|▋| 259/370 [01:53checkpoint...
Step: 112109. Epoch: 302/500. Iteration: 370/370. Cls loss: 0.49243. Reg loss: 0.08511. Total loss: 0.57754: 100%|█| 370/370 [02:39
Val. Epoch: 302/500. Classification loss: 0.56575. Regression loss: 0.10168. Total loss: 0.66743
Step: 112479. Epoch: 303/500. Iteration: 370/370. Cls loss: 0.51703. Reg loss: 0.06826. Total loss: 0.58529: 100%|█| 370/370 [02:38
Val. Epoch: 303/500. Classification loss: 0.56428. Regression loss: 0.09985. Total loss: 0.66414
Step: 112499. Epoch: 304/500. Iteration: 20/370. Cls loss: 0.49051. Reg loss: 0.08012. Total loss: 0.57063:   5%| | 19/370 [00:12<0checkpoint...
Step: 112849. Epoch: 304/500. Iteration: 370/370. Cls loss: 0.71797. Reg loss: 0.07365. Total loss: 0.79162: 100%|█| 370/370 [02:39
Val. Epoch: 304/500. Classification loss: 0.56507. Regression loss: 0.10104. Total loss: 0.66612
![image](https://user-images.githubusercontent.com/56627406/80853365-da936d80-8c62-11ea-823f-8927d67fe0a8.png)

**Comment by xiaoshuyun** [comment]
Can you help me . I'm looking forward to your reply.thanks for your time

**Comment by kuangsangudu** [comment]
me too, i don't know why

**Comment by zylo117** [comment]
how many classes in your dataset?

**Comment by xiaoshuyun** [comment]
> 您的数据集中有几类？

only one class : ship
thanks for your time

**Comment by zylo117** [comment]
is the category id 1, not 0?

**Comment by xiaoshuyun** [comment]
> is the category id 1, not 0?

yes.I convert .xml to .json.
![image](https://user-images.githubusercontent.com/56627406/80913333-74980a80-8d76-11ea-8146-f033f98b07c1.png)

**Comment by zylo117** [comment]
but the loss was dropping at the early stage, right? It just stagnated after a few epoch?

try `--optim sgd` or smaller lr.

**Comment by xiaoshuyun** [comment]
> but the loss was dropping at the early stage, right? It just stagnated after a few epochs?
> 
> try `--optim sgd` or smaller lr.

about 100 epoch.  OK,I will try it

**Comment by xiaoshuyun** [comment]
> but the loss was dropping at the early stage, right? It just stagnated after a few epochs?
> 
> try `--optim sgd` or smaller lr.

By the time , I hope to know that How many pictures are needed to train in general?for example,only one class?

**Comment by zylo117** [comment]
Depends. But in general, the more the better.

**Comment by xielaoban24** [comment]
Hello, can you share your python script that converts .csv to .json please? I always have some problems,thank you very much!  my email 1257512279@qq.com

**Comment by xiaoshuyun** [comment]
> Hello, can you share your python script that converts .csv to .json please? I always have some problems,thank you very much! my email [1257512279@qq.com](mailto:1257512279@qq.com)

my code is to convert .xml to .json，maybe it is not fit for you?

**Comment by ChrisLiiiii** [comment]
> > Hello, can you share your python script that converts .csv to .json please? I always have some problems,thank you very much! my email [1257512279@qq.com](mailto:1257512279@qq.com)
> 
> my code is to convert .xml to .json，maybe it is not fit for you?

Could you please share your script to convert .xml to .json? That may be helpful to me... My email 453055043@qq.com. Thanks a lot!

**Comment by xiaoshuyun** [comment]
> > > Hello, can you share your python script that converts .csv to .json please? I always have some problems,thank you very much! my email [1257512279@qq.com](mailto:1257512279@qq.com)
> > 
> > 
> > my code is to convert .xml to .json，maybe it is not fit for you?
> 
> Could you please share your script to convert .xml to .json? That may be helpful to me... My email [453055043@qq.com](mailto:453055043@qq.com). Thanks a lot!

https://github.com/JialianW/pascal2coco

**Comment by ChrisLiiiii** [comment]
> > > > Hello, can you share your python script that converts .csv to .json please? I always have some problems,thank you very much! my email [1257512279@qq.com](mailto:1257512279@qq.com)
> > > 
> > > 
> > > my code is to convert .xml to .json，maybe it is not fit for you?
> > 
> > 
> > Could you please share your script to convert .xml to .json? That may be helpful to me... My email [453055043@qq.com](mailto:453055043@qq.com). Thanks a lot!
> 
> https://github.com/JialianW/pascal2coco

thanks again!

**Comment by xiaoshuyun** [comment]
hello~do you get good results?would you like communicate with me? thank you for your time.

**Comment by ChrisLiiiii** [comment]
> hello~do you get good results?would you like communicate with me? thank you for your time.

Sorry, I've been doing segmentation recently, I'll try this later.
